.github/workflows/release.yaml

```
@@ -0,0 +1,193 @@
+          else
+            echo "Invalid release type: ${{ inputs.version-type }}"
+            exit 1
+          fi
+
+          LAST_MATCH_TAG=$(git tag -l --sort=-v:refname "$TAG_QUERY" | head -n 1)
+
+          if [ "${{ inputs.version-type }}" == "GA" ] && [ "$LAST_MATCH_TAG" != "" ]; then
+            echo "Release $LAST_MATCH_TAG already exists"
+            exit 1
+          fi
+
+          if [ "${{ inputs.version-type }}" == "alpha" ] || [ "${{ inputs.version-type }}" == "beta" ] ; then
+            SUFFIX_NUMBER="1"
+
+            if [ "$LAST_MATCH_TAG" != "" ]; then
+              SUFFIX_NUMBER=$(./semver get prerel $LAST_MATCH_TAG | cut -d'.' -f2)
+              SUFFIX_NUMBER=$((SUFFIX_NUMBER + 1))
+            fi
+
+            TAG="$TAG.$SUFFIX_NUMBER"
+          fi
+
+          echo "SHORT_VERSION=$SHORT_VERSION" >> $GITHUB_ENV
+          echo "PATCH_VERSION=$VERSION_PATCH" >> $GITHUB_ENV
+          echo "BASE_VERSION=$BASE_VERSION" >> $GITHUB_ENV
+          echo "RELEASE_TAG=$TAG" >> $GITHUB_ENV
+          echo "RELEASE_BRANCH=release/$TAG" >> $GITHUB_ENV
+      - name: Validate ${{ env.RELEASE_TAG }} tag
+        run: ./semver validate ${{ env.RELEASE_TAG }}
+      - name: Check if version ${{ env.BASE_VERSION }} is on the right branch
+        run: |
+          # Need to fetch dev branches first
+          git fetch --no-tags origin HEAD '+refs/heads/development/*:refs/remotes/origin/development/*'
+
+          GIT_REMOTE=$(git remote)
+          BASE_BRANCH=$(git branch -a --list ${GIT_REMOTE}/development/${{ env.SHORT_VERSION }})
+          if ! git merge-base --is-ancestor HEAD ${BASE_BRANCH}; then
+            echo "Version commit is not included in base branch ${BASE_BRANCH}"
+            exit 1
+          fi
+      - name: Prepare branch
+        run: git checkout -b ${{ env.RELEASE_BRANCH }}
+      - name: Set the new version
+        run: |
+          if [ -z "${NEW_VERSION_SUFFIX}" ]; then
+            NEW_VERSION_SUFFIX=$(./semver get prerel ${{ env.RELEASE_TAG }})
+            [ -n "$NEW_VERSION_SUFFIX" ] && NEW_VERSION_SUFFIX="-$NEW_VERSION_SUFFIX"
+          fi
+          sed -i "s/VERSION_SUFFIX=.*/VERSION_SUFFIX=$NEW_VERSION_SUFFIX/" VERSION
+      - name: Commit with requested version and push
+        run: |
+          git fsck
+          git gc
+          git add VERSION
+          git config --global user.email ${{ github.actor }}@scality.com
+          git config --global user.name ${{ github.actor }}
+          git commit -m "Release ${{ env.RELEASE_TAG }}"
+          git push --set-upstream origin ${{ env.RELEASE_BRANCH }}
+      - name: Create and push `${{ env.RELEASE_TAG }}` tag
+        run: |
+          BODY=$(mktemp)
+
+          PATCH_MSG=""
+          if [ "${{ env.PATCH_VERSION }}" != "0" ]; then
+            PATCH_MSG="MetalK8s ${{ env.RELEASE_TAG }} is a patch release (see [the main release ${{ env.BASE_VERSION }}](https://github.com/scality/metalk8s/releases/${{ env.BASE_VERSION }}))."
```

No base version is just the "previous" version, here we want the first patch version

```suggestion
            PATCH_MSG="MetalK8s ${{ env.RELEASE_TAG }} is a patch release (see [the main release ${{ env.SHORT_VERSION }}.0](https://github.com/scality/metalk8s/releases/${{ env.SHORT_VERSION }}.0))."
```

---

Done!